### PR TITLE
Enable authentication between go and mongo db

### PR DESCRIPTION
**Pull request by ydrisR**
* Change mgo source package from labix.org/v2/mgo to gopkg.in/mgo.v2.
    Not sure why it is important as this is suposed to be the same code... but this is the only way I found to make the connection to mongo work (using sasl)
    * change the docker file to build the mgo module with SASL support. This requires the SASL library + a bunch of build tools (gcc)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,18 +1,22 @@
 # Development
-FROM golang:1.9.2-alpine AS development
+FROM golang:1.10.2-alpine AS development
```
Comment (franck-fourel) [inline Dockerfile:2]: is it necessary to upgrade to 1.10.2? 
Travis is still configured with 1.9.2

Comment (ydrisR) [reply Dockerfile:2]: Not necessary at all: I simply did that to be aligned with what is done on the platform service.
1.9.2 was release more that a year ago, I supose we should try to keep up to date as much as we can.

**Comment by franck-fourel** [comment]
I do not squash the merge so that we can cherry pick the ca602e2 to create a pull request on tidepool.

**Comment by ydrisR** [comment]
@franck-fourel Travis is happy now ;)

**Pull request merged**